### PR TITLE
Target SDK should attach response tokens in the retrieveLocationContent data callback

### PR DESCRIPTION
**Pull request by swarna04**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPTarget/Tests/TestHelpers/NamedCollectionDataStore+clear.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPServices
+import Foundation
+
+extension NamedCollectionDataStore {
+    static func clear(appGroup: String? = nil) {
```
Comment (sbenedicadb) [inline AEPTarget/Tests/TestHelpers/NamedCollectionDataStore+clear.swift:17]: this is useful across all our extensions - have you talked with @timkimadobe about getting it added to our test-utils framework?

File: AEPTarget/Tests/TestHelpers/UserDefaults+clear.swift [hunk 2]
```diff
@@ -12,10 +12,20 @@
...
 extension UserDefaults {
-    static func clear() {
-        for _ in 0 ... 5 {
-            for key in UserDefaults.standard.dictionaryRepresentation().keys {
-                UserDefaults.standard.removeObject(forKey: key)
+    static func clear(appGroup: String? = nil) {
+        if let appGroup = appGroup,
+           !appGroup.isEmpty,
+           let userDefaults = UserDefaults(suiteName: appGroup) {
+                for _ in 0 ... 5 {
```
Comment (sbenedicadb) [inline AEPTarget/Tests/TestHelpers/UserDefaults+clear.swift:19]: do we have a better way to iterate through defaults to clear them?  curious as to why we are using a for loop here.

**Review by sbenedicadb — COMMENTED** [review]
looks good - couple questions we can chat about on the side.  

probably a good idea to run the UpdateVersions GH action to ensure we didn't miss anything

File: AEPTarget/Tests/TestHelpers/UserDefaults+clear.swift (hunk 2 above)
Comment (swarna04) [reply AEPTarget/Tests/TestHelpers/UserDefaults+clear.swift:19]: This code has been around for a bit to ensure the defaults get cleared. You can find it elsewhere too, see https://github.com/adobe/aepsdk-core-ios/blob/main/AEPCore/Mocks/UserDefaults%2BClear.swift.

**Review by sbenedicadb — APPROVED** [review]

**Pull request merged**